pipelines/internal/commands/run/run.go

```diff
@@ -133,6 +133,13 @@ import (
 	"google.golang.org/api/googleapi"
 )
 
+type LocaleType int
```

Comment:
There's no need to create an enum and pass it around.  You can use a function instead:

var resources *genomics.Resources
if len(*zones) > 0 {
  zones, err := expandPrefixes(project, listOf(*zones), listZones)
  if err... {}
  resources = &genomics.Resources{
    Zones: zones,
  }
} else {
  if len(*regions) == 0 {
    // Default region
  }
  // Regions as above
}

// expandZones -> expandPrefixes
func expandPrefixes(project string, input []string, allValues func(project string, service *compute.Service) ([]string, error) {
  // Move service creation into expandPrefixes so it's shared and call allValues.
}

func listZones(project string, service *compute.Service) ([]string, error) {
  resp, err := service.Zones.List(project).Do()
  if err != nil {
    return nil, fmt.Errorf("listing zones: %v", err)
  }
  var zones []string
  for _, zone := range resp.Items {
    zones = append(zones, zone.Name)
  }
  return zones, nil
}

func listRegions ...

Reply:
Done.

Not sure if it's better to use *zones != "" vs len(*zones) > 0, but followed your suggestion.

Slightly modified your suggestion to extract default value for zone directly from the flag's definition rather than hardcoding it in the if statement - this way, zone will be supplied one way or another, so no if is necessary.